ibis/backends/flink/utils.py

```diff
@@ -0,0 +1,232 @@
+import datetime
+import math
+from abc import ABC, abstractmethod
+from collections import defaultdict
+
+from ibis.common.temporal import IntervalUnit
+from ibis.util import convert_unit
+
+MIN_ALLOWED_PRECISION = {
```

Comment:
is precision an ibis or flink concept? if the later, we should add some comments for clarity (i.e., what does each precision mean).